index.js

```diff
@@ -1,23 +1,161 @@
-const { ApolloServer, gql }   = require('apollo-server')
-const dbResolvers             = require('./db')
-const apiResolvers            = require('./api')
+const { ApolloServer, gql } = require('apollo-server')
+const { GraphQLScalarType, Kind } = require('graphql')
+
+// const { startCron } = require('./src/cron')
+const dbResolvers = require('./src/resolvers/db/index')
+const apiResolvers = require('./src/resolvers/api/index')
 
 const typeDefs = gql`
+
+  scalar Date
+
+  enum StatusName {
+    LOCKED
+    UNLOCKED
+    SPAWNED
+    ACTIVATED
+  }
+
+  enum NodeType {
+    GALAXY
+    STAR
+    PLANET
+    # MOON
+    # COMET
+  }
+
+  enum EventName {
+    CHANGE_OWNERSHIP
+    CHANGE_SPAWN_PROXY
+    CHANGE_TRANSFER_PROXY
+    CHANGE_MANAGEMENT_PROXY
+    CHANGE_VOTING_PROXY
+    ACTIVATE
+    SPAWN
+    ESCAPE_REQUESTED
+    ESCAPE_CANCELLED
+    ESCAPE_ACCEPTED
+    LOST_SPONSOR
+    BROKE_CONTINUITY
+  }
+
+  input GetNodeInput {
+    urbitId: String
+  }
+
+  input GetNodesInput {
+    q: String
+    nodeTypes: [NodeType]
+    limit: Int
+    offset: Int
+  }
+
+  input PKIEventInput {
+    urbitId: String
+    since: Date
+    nodeTypes: [NodeType]
+    limit: Int
+    offset: Int
+  }
+
+  input GetActivityInput {
+    urbitId: String
+    since: Date
+    until: Date
+  }
+
+  type NodeStatus {
+    nodeStatusId: Int!
+    statusName: StatusName!
+  }
+  
+  type Node {
+    urbitId: String
+    numOwners: Int
+    sponsors: [String]
```

Comment:
In the [bounty description](https://grants.urbit.org/bounties/1101665172), `get-node` returns `sponsors: [node] (sponsor and sponsor's sponsor, if applicable)`. In that case, this should be an array. Let me know if I am misunderstanding something.

Reply:
Ah, good catch. I'm okay with changing it to a single `Node` and having the client get the sponsor's sponsor via GraphQL (there's another comment in this review that references this), or keeping it as an array. Regardless though, the type should be a Node and not a String.

Reply:
If it's all the same, I'd prefer keeping it as an array as this requires me to redo less. I have changed it to `Node` from `String`, though.

Reply:
After digging deeper into constructing a materialized view for `node`, it is in fact a better approach to use it as a single `Node` instead of an array, so that is what I will do.